system/sensord/rawgps/rawgpsd.py

```diff
@@ -120,13 +122,37 @@ def setup_quectel(diag: ModemDiag):
 
   if gps_enabled():
     at_cmd("AT+QGPSEND")
+  #at_cmd("AT+QGPSDEL=0")
 
   # disable DPO power savings for more accuracy
   at_cmd("AT+QGPSCFG=\"dpoenable\",0")
   # don't automatically turn on GNSS on powerup
   at_cmd("AT+QGPSCFG=\"autogps\",0")
 
-  at_cmd("AT+QGPSSUPLURL=\"supl.google.com:7275\"")
+  # Do internet assistance
+  at_cmd("AT+QGPSXTRA=1")
+  assist_data_file = '/tmp/xtra3grc.bin'
+  assistance_url = 'http://xtrapath3.izatcloud.net/xtra3grc.bin'
+  try:
+    with open(assist_data_file, "wb") as fp:
+      curl = pycurl.Curl()
+      curl.setopt(pycurl.URL, assistance_url)
+      curl.setopt(pycurl.CONNECTTIMEOUT, 5)
+
+      curl.setopt(pycurl.WRITEDATA, fp)
+      curl.perform()
+      curl.close()
+  except pycurl.error as e:
+    cloudlog.exception(f'Failed to download assistance file with error: {e}')
+  if os.path.isfile(assist_data_file):
+    try:
+      subprocess.check_call(f"mmcli -m any --timeout 30 --location-inject-assistance-data={assist_data_file}", shell=True)
+    except subprocess.CalledProcessError:
+      cloudlog.exception("rawgps.mmcli_command_failed")
+  #at_cmd("AT+QGPSXTRADATA?")
+  time_str = datetime.utcnow().strftime("%Y/%m/%d,%H:%M:%S")
+  at_cmd(f"AT+QGPSXTRATIME=0,\"{time_str}\",1,1,1000")
```

Comment:
docs say time has to go first. perhaps ModemManager is already doing this for us in `--location-inject-assistance-data`?

Reply:
I tested both ways seemed the same. And no modemmanager does not not do it for us